lec15-git-projects.Rmd

```diff
@@ -0,0 +1,415 @@
+instead.
+
+![_Pictured: the branch dropdown menu_](image/branch_dropdown.png)
+
+We can now use the dropdown to switch back to `master`, but let's stay
+in `initial-edit` for now. This is the place to make new changes -- 
+either by editing existing files or creating new ones. Let's create
+a simple Rmd file with the 'Create new file' button on the right side
+of the repo page.
+
+````
+---
+title: "Test R Notebook"
+author: "Your Name"
+output: pdf_document
+---
+
+Hello! This is my R notebook.
+
+```{r}`r ''`
+print('Hello, GitHub!')
+```
+
+````
+
+![_Pictured: editing a file on GitHub_](image/sample_rmd.png)
+
+Once done, scroll to the bottom of the page and add a commit message 
+using the text file -- something like 'initial Rmd file creation' --
+and then click on 'Commit new file'. 
+
+#### Pull requests
+
+After we've made our commit, GitHub takes us back to the main repo
+page, but now with an attention-grabbing yellow prompt about our
+new edit. GitHub has noticed that we have a new branch with
+edits, and over on the right side of the prompt, we have the option
+of creating a pull request.
+
+![_Pictured: GitHub's pull request prompt_](image/yellow_prompt.png)
+
+If you don't see the yellow prompt, head to your fork and switch to the
+branch containing your edits. Underneath the green 'Clone or download' 
+button, there is a 'Pull request' option -- this can be used to open
+a pull request instead. You'll also notice that the left side of this
+bar lists how many commits ahead of `master` your branch currently is.
+
+Either way, once we decide to open a pull request, GitHub takes us to
+the 'Pull requests' tab of the _main repo_ and prompts us to write about
+our pull request. Here, we can (and should) explain the changes we've
+made for our group members, so that they know what to look for and review.
+Be specific and detailed to save your group members' time -- it's a good
+idea to start off your pull request message with an overall summary 
+('adding data cleaning `dplyr` code') followed by a point-form list of
+what changes have been made, if necessary. 
+
+Once the pull request has been made, GitHub will list both your message
+and your commit messages below. You also have the option of merging the
+pull request yourself -- but _don't do this_! When collaborating, always
+have someone else review and merge your pull request. 
+
+If all does not look good, your team members can add messages below, and
+tag others using the @ symbol, similarly to most social networks. If
+more changes are needed before the pull request is ready to merge, 
+any new commits you make to the branch on your fork (`initial-edit` in
+our case here) will automatically be added on to the pull request.
+This way, you can incorporate any changes or fixes suggested by your
+team members simply by continuing to work in your fork's new branch
+until your changes are ready to merge.
+
+Once a pull request has been merged, the branch on our fork isn't needed
```

Comment:
Merged where? Should mention explicitly that the branch is being merged into `master` branch of the main repo.